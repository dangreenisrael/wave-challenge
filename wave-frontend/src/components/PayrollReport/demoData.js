export default [
  {
    employeeId: 1,
    payPeriod: '01/11/2016 - 15/11/2016',
    totalPay: 150
  },
  {
    employeeId: 2,
    payPeriod: '01/11/2016 - 15/11/2016',
    totalPay: 930
  },
  {
    employeeId: 3,
    payPeriod: '01/11/2016 - 15/11/2016',
    totalPay: 590
  },
  {
    employeeId: 4,
    payPeriod: '01/11/2016 - 15/11/2016',
    totalPay: 150
  },
  {
    employeeId: 1,
    payPeriod: '16/11/2016 - 30/11/2016',
    totalPay: 220
  },
  {
    employeeId: 4,
    payPeriod: '16/11/2016 - 30/11/2016',
    totalPay: 450
  },
  {
    employeeId: 1,
    payPeriod: '01/12/2016 - 15/12/2016',
    totalPay: 150
  },
  {
    employeeId: 2,
    payPeriod: '01/12/2016 - 15/12/2016',
    totalPay: 930
  },
  {
    employeeId: 3,
    payPeriod: '01/12/2016 - 15/12/2016',
    totalPay: 470
  },
  {
    employeeId: 4,
    payPeriod: '01/12/2016 - 15/12/2016',
    totalPay: 150
  },
  {
    employeeId: 1,
    payPeriod: '16/12/2016 - 31/12/2016',
    totalPay: 220
  },
  {
    employeeId: 4,
    payPeriod: '16/12/2016 - 31/12/2016',
    totalPay: 450
  },
  {
    employeeId: 4,
    payPeriod: '16/02/2015 - 28/02/2015',
    totalPay: 100
  },
  {
    employeeId: 4,
    payPeriod: '16/02/2016 - 29/02/2016',
    totalPay: 150
  }
];
