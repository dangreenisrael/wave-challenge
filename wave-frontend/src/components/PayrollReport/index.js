import React, {Component} from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import {BootstrapTable, TableHeaderColumn} from 'react-bootstrap-table';
import 'react-bootstrap-table/dist/react-bootstrap-table-all.min.css';

const getGetPayPeriodSortable = ({payPeriod}) => {
  const [startDate] = payPeriod.split(' - ');
  return moment(startDate, 'DD-MM-YYYY').unix();
};
const sortByPayPeriod = (a, b, order) => {
  const A = getGetPayPeriodSortable(a);
  const B = getGetPayPeriodSortable(b);
  if (order === 'desc') {
    return A - B;
  } else {
    return B - A;
  }
};
class PayrollReport extends Component {
  render() {
    const {data} = this.props;
    const options = {
      defaultSortName: 'employeeId', // default sort column name
      defaultSortOrder: 'acs' // default sort order,
    };
    if (!data[0]) return <h3>Please upload a time report</h3>;
    return (
      <BootstrapTable ref="table" data={data} multiColumnSort={2} options={options}>
        <TableHeaderColumn isKey dataField="employeeId" dataSort>
          Employee Id
        </TableHeaderColumn>
        <TableHeaderColumn dataField="payPeriod" dataSort sortFunc={sortByPayPeriod}>
          Pay Period
        </TableHeaderColumn>
        <TableHeaderColumn dataField="totalPay" dataSort>
          Total Pay
        </TableHeaderColumn>
      </BootstrapTable>
    );
  }
}

PayrollReport.propTypes = {
  data: PropTypes.array.isRequired
};

export {PayrollReport as default, getGetPayPeriodSortable, sortByPayPeriod}; // Exported for testability
