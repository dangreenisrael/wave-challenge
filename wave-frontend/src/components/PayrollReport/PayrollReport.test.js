import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import PayrollReport from '.';
import data from './__testData__/payrollReportData';

test('PayrollReport Renders Correctly', () => {
  const wrapper = shallow(<PayrollReport data={data} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
