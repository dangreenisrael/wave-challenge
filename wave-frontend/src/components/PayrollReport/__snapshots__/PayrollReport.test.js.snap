// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PayrollReport Renders Correctly 1`] = `
<BootstrapTable
  bodyContainerClass={null}
  bordered={true}
  cellEdit={
    Object {
      "afterSaveCell": undefined,
      "beforeSaveCell": undefined,
      "blurToSave": false,
      "mode": "none",
      "nonEditableRows": undefined,
    }
  }
  columnFilter={false}
  condensed={false}
  containerClass={null}
  csvFileName="spreadsheet.csv"
  data={
    Array [
      Object {
        "employeeId": 1,
        "payPeriod": "01/11/2016 - 15/11/2016",
        "totalPay": 150,
      },
      Object {
        "employeeId": 2,
        "payPeriod": "01/11/2016 - 15/11/2016",
        "totalPay": 930,
      },
      Object {
        "employeeId": 3,
        "payPeriod": "01/11/2016 - 15/11/2016",
        "totalPay": 590,
      },
      Object {
        "employeeId": 4,
        "payPeriod": "01/11/2016 - 15/11/2016",
        "totalPay": 150,
      },
      Object {
        "employeeId": 1,
        "payPeriod": "16/11/2016 - 30/11/2016",
        "totalPay": 220,
      },
      Object {
        "employeeId": 4,
        "payPeriod": "16/11/2016 - 30/11/2016",
        "totalPay": 450,
      },
      Object {
        "employeeId": 1,
        "payPeriod": "01/12/2016 - 15/12/2016",
        "totalPay": 150,
      },
      Object {
        "employeeId": 2,
        "payPeriod": "01/12/2016 - 15/12/2016",
        "totalPay": 930,
      },
      Object {
        "employeeId": 3,
        "payPeriod": "01/12/2016 - 15/12/2016",
        "totalPay": 470,
      },
      Object {
        "employeeId": 4,
        "payPeriod": "01/12/2016 - 15/12/2016",
        "totalPay": 150,
      },
      Object {
        "employeeId": 1,
        "payPeriod": "16/12/2016 - 31/12/2016",
        "totalPay": 220,
      },
      Object {
        "employeeId": 4,
        "payPeriod": "16/12/2016 - 31/12/2016",
        "totalPay": 450,
      },
      Object {
        "employeeId": 4,
        "payPeriod": "16/02/2015 - 28/02/2015",
        "totalPay": 100,
      },
      Object {
        "employeeId": 4,
        "payPeriod": "16/02/2016 - 29/02/2016",
        "totalPay": 150,
      },
    ]
  }
  deleteRow={false}
  expandColumnOptions={
    Object {
      "expandColumnBeforeSelectColumn": true,
      "expandColumnComponent": undefined,
      "expandColumnVisible": false,
    }
  }
  exportCSV={false}
  fetchInfo={
    Object {
      "dataTotalSize": 0,
    }
  }
  headerContainerClass={null}
  height="100%"
  hover={false}
  ignoreSinglePage={false}
  insertRow={false}
  keyBoardNav={false}
  multiColumnSearch={false}
  multiColumnSort={2}
  options={
    Object {
      "defaultSortName": "employeeId",
      "defaultSortOrder": "acs",
    }
  }
  pagination={false}
  printable={false}
  search={false}
  selectRow={
    Object {
      "bgColor": "",
      "clickToExpand": false,
      "clickToSelect": false,
      "clickToSelectAndEditCell": false,
      "customComponent": undefined,
      "hideSelectColumn": false,
      "mode": "none",
      "onSelect": undefined,
      "onSelectAll": undefined,
      "onlyUnselectVisible": false,
      "selected": Array [],
      "showOnlySelected": false,
      "unselectable": Array [],
    }
  }
  striped={false}
  tableBodyClass={null}
  tableContainerClass={null}
  tableHeaderClass={null}
  trClassName=""
>
  <TableHeaderColumn
    className=""
    columnClassName=""
    columnTitle={false}
    dataAlign="left"
    dataField="employeeId"
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filterFormatted={false}
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={true}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width={null}
  >
    Employee Id
  </TableHeaderColumn>
  <TableHeaderColumn
    className=""
    columnClassName=""
    columnTitle={false}
    dataAlign="left"
    dataField="payPeriod"
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filterFormatted={false}
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={false}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width={null}
  >
    Pay Period
  </TableHeaderColumn>
  <TableHeaderColumn
    className=""
    columnClassName=""
    columnTitle={false}
    dataAlign="left"
    dataField="totalPay"
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filterFormatted={false}
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={false}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width={null}
  >
    Total Pay
  </TableHeaderColumn>
</BootstrapTable>
`;
